internal/batch/service.go

```diff
@@ -0,0 +1,112 @@
+package batch
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	goabatch "github.com/artefactual-labs/enduro/internal/api/gen/batch"
+	"github.com/artefactual-labs/enduro/internal/cadence"
+	"github.com/artefactual-labs/enduro/internal/collection"
+	"github.com/artefactual-labs/enduro/internal/validation"
+	"github.com/go-logr/logr"
+	"go.uber.org/cadence/.gen/go/shared"
+	cadenceclient "go.uber.org/cadence/client"
+)
+
+var ErrBatchStatusUnavailable = errors.New("batch status unavailable")
+
+//go:generate mockgen  -destination=./fake/mock_batch.go -package=fake github.com/artefactual-labs/enduro/internal/batch Service
+
+type Service interface {
+	Submit(context.Context, *goabatch.SubmitPayload) (res *goabatch.BatchResult, err error)
+	Status(context.Context) (res *goabatch.BatchStatusResult, err error)
+	InitProcessingWorkflow(ctx context.Context, batchDir, key, pipelineName string) error
+}
```

Comment:
We should eventually revisit this pattern of wide interfaces - I've set the precedent but I'm willing to challenge it. Its only value at the moment is to help us test the batch activity, where we're only using InitProcessingWorkflow. Maybe `Submit` and `Status` should go if this interface doesn't have a consumer using them.